server/middleware/validationMiddleware.ts

```diff
@@ -1,24 +1,32 @@
 import { plainToInstance } from 'class-transformer'
 import { validate, ValidationError } from 'class-validator'
-import { RequestHandler } from 'express'
-import ConditionService from '../services/conditionService'
+import type { RequestHandler } from 'express'
+import type { Licence } from '../@types/licenceApiClientTypes'
+import type ConditionService from '../services/conditionService'
+import asyncMiddleware from './asyncMiddleware'
 
 export type FieldValidationError = {
   field: string
   message: string
 }
 
+async function getValidator(
+  conditionService: ConditionService,
+  licenceVersion: string | undefined,
+  conditionCode: string,
+  type: new () => object
+): Promise<new () => object> {
+  if (!conditionCode) return type
+  const condition = await conditionService.getAdditionalConditionByCode(conditionCode, licenceVersion)
+  return condition?.validatorType || type
+}
+
 function validationMiddleware(conditionService: ConditionService, type?: new () => object): RequestHandler {
-  return async (req, res, next) => {
-    const { licence } = res.locals
+  return asyncMiddleware(async (req, res, next) => {
```

Comment:
wrapping in async middleware which will forward any unexpected async errors to the next function (and so pipe through to the error handler)